### PR TITLE
Don't map uid/gid values by user/group name (--numeric-ids)

### PR DESCRIPTION
**Pull request by usrflo**
The documentation comprises grid/group (for compat. reasons). In the next step there will be a pull request based on the numeric-ids branch to merge in the gid/group extension.
The system test named "testClientCopyPreserveUid" requires to be executed as 'root'. I propose to just skip the test when user!=root. Currently the test fails if superuser privileges are not available.
Scripts like http://stackoverflow.com/questions/2580279/how-do-i-run-my-application-as-superuser-from-eclipse simplifies the execution as root in an IDE.

### CODE REVIEW

**Comment by perlundq** [comment]
Style wise the patch looks really good. But it has a severe issue when being run as a server and receiving --numeric-ids from a peer client. E.g.: 
yajsyncd with a module called downloads where testfile is owned by a user that does not exist on the client side
```
rsync -o --numeric-ids rsync://192.168.122.1:14415/downloads/testfile .
```
results in:
```
File-list index 130 not in -1 - 0 (read_ndx_and_attrs) [receiver]
rsync error: protocol incompatibility (code 2) at rsync.c(654) [receiver=3.0.9]
rsync: connection unexpectedly closed (28 bytes received so far) [generator]
rsync error: error in rsync protocol data stream (code 12) at io.c(605) [generator=3.0.9]
```

- Please run git diff --check also before committing (and make sure you have: git config core.whitespace trailing-space,space-before-tab,tab-in-indent,tabwidth=4)
- And yes, please only run that test if actually being root.

File: src/main/com/github/perlundq/yajsync/session/Sender.java [hunk 1]
```diff
@@ -276,7 +285,8 @@ public Boolean call() throws ChannelException, InterruptedException
             }
             long t3 = System.currentTimeMillis();
 
-            if (_isPreserveUser && _fileSelection != FileSelection.RECURSE) {
+            if (_isPreserveUser && !_isNumericIds
+                    && _fileSelection != FileSelection.RECURSE) {
                 sendUserList();
```
Comment (usrflo) [inline src/main/com/github/perlundq/yajsync/session/Sender.java]: Regarding the sent error message from the native rsync receiver (list entry not found) I guess if the sending of the user list should be done although the user data is not required on the receiver side. Do you have another assumption? I didn't check the rsync source code yet.

Comment (perlundq) [inline src/main/com/github/perlundq/yajsync/session/Sender.java]: sorry, but no, you really have to look into the rsync code to find out

Comment (usrflo) [inline src/main/com/github/perlundq/yajsync/session/Sender.java]: I fixed this problem with commit https://github.com/usrflo/yajsync/commit/93d508d4f2be03e1895a7b0323baad12e8666139 but failed to squash the two commits of this branch (maybe you can add these 3 lines in your submit directly).

**Comment by perlundq** [comment]
It still does not work properly. Please make sure you test your code thoroughly before submitting a pull request. In this case it does not work when sending files (same case as above).
The reasons for this are:
- In RsyncClient.Send if being remote sender, it should set isNumericIds otherwise you'll get an protocol error
- In Sender.sendFileMetaData: Avoid setting TransmitFlags.USER_NAME_FOLLOWS if numeric-ids (this is not obvious unless you look at the rsync source)
- Make that test pass if not running as root.

For squashing a commit you use
 git rebase -i
while also specifying the range of commits to squash. See for example http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request

**Comment by perlundq** [comment]
> In RsyncClient.Send if being remote sender, it should set isNumericIds otherwise you'll get an protocol error

I do not know why I missed this. But you had already fixed it, sorry.

> In Sender.sendFileMetaData: Avoid setting TransmitFlags.USER_NAME_FOLLOWS if numeric-ids 

I've added a fix for this. The problem is when numeric-ids is requested the usernames would be sent anyway.

Regarding the other stuff I've added some minor fixes.
Thanks!

**Pull request merged**